dashboard/app/models/course_version.rb

```diff
@@ -189,10 +189,10 @@ def recommended?(locale_code = 'en-us')
   # CSD has multiple headers in the list with the units for that year under it.
   # See fakeCoursesWithProgress in teacherDashboardTestHelpers.js for an example of what
   # the resulting data looks like
-  def self.courses_for_unit_selector(unit_ids)
+  def self.courses_for_unit_selector(unit_ids, user)
     standalone_units = Unit.joins(:course_version).where(id: unit_ids).map {|u| u.course_version.course_offering&.summarize_for_unit_selector(unit_ids)}.compact.uniq
 
-    unit_groups =  Unit.joins(unit_groups: :course_version).where(id: unit_ids).where(course_version: {content_root_type: 'UnitGroup'}).flat_map {|u| u.unit_groups.map(&:course_version)}.map(&:summarize_for_unit_selector).uniq
+    unit_groups =  Unit.joins(unit_groups: :course_version).where(id: unit_ids).where(course_version: {content_root_type: 'UnitGroup'}).flat_map {|u| u.unit_groups.map(&:course_version)}.select {|cv| cv.course_assignable?(user)}.map(&:summarize_for_unit_selector).uniq
```

Comment:
Calling `.course_assignable?` directly on a CourseVersion to filter them in the results rather than calling it on a `Unit`

Reply:
sounds great! we should be getting rid of `unit.course_assignable?` soon.